data/base/sub-controller.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Elementor\Data\Base;
+
+use Elementor\Data\Manager;
+
+abstract class SubController extends Controller {
+	/**
+	 * @var \Elementor\Data\Base\Controller
+	 */
+	public $parent_controller;
+
+	/**
+	 * SubController constructor.
+	 *
+	 * $parent_controller is optional, if not passed will use `$this->get_parent_name()`.
+	 *
+	 * @param null|\Elementor\Data\Base\Controller $parent_controller
+	 */
+	public function __construct( $parent_controller = null ) {
+		$this->parent_controller = $parent_controller;
+
+		if ( ! $this->parent_controller ) {
+			$this->parent_controller = Manager::instance()->get_controller( $this->get_parent_name() );
+		}
+
+		if ( ! $this->parent_controller ) {
+			trigger_error( 'Cannot find parent controller' );
+			return;
+		}
+
+		if ( $this->parent_controller instanceof SubController ) {
+			trigger_error( 'Parent cannot be sub-controller use endpoints/sub-endpoints' );
+			return;
+		}
+
+		parent::__construct();
+	}
+
+	public function get_full_name() {
+		return $this->parent_controller->get_name() . '/' . parent::get_full_name();
+	}
+
+	public function get_rest_base() {
+		$route = $this->get_route();
```

Comment:
An empty route should  return a `/`. so you don't need add slashes.

Reply:
Handled https://github.com/elementor/elementor/pull/12820/commits/b4cdcd462691d96b8f0b3424c8e636315098bab2